### ISSUE TITLE
M-q does not work correctly inside comment block (javascript)

### ISSUE THREAD

**Issue by miguel-guedes** [fxbois/web-mode#733]
Given the following comment sample comment block:

``` javascript
/* A rather long comment spanning longer than 80 characters that talks about radioactive leprechauns */
```

Assuming point is anywhere inside the comment block, <kbd>M-q</kbd> should format the comment block nicely, but instead the following happens:

``` javascript
/* A rather long comment spanning longer than 80 characters that talks about
radioactive leprechauns */
```

Expected:

``` javascript
/* A rather long comment spanning longer than 80 characters that talks about
 * radioactive leprechauns */
```

**Comment by bmalehorn** [comment]
This issue bothers me, too, so I looked into it.

The goal is to call `c-fill-paragraph`, which does the fancy `*` thing we've come to expect in C/C++/Java. These major modes all derive from `cc-mode`.

However, calling `c-fill-paragraph` from `web-mode` doesn't work. It only works if called from within `cc-mode`.

So a gross workaround is:

    (defun my-fill-paragraph ()
      (c-mode)
      (fill-paragraph)
      (web-mode))

The real solution is to read the source on `c-fill-paragraph` and trick it into running outside of `cc-mode`.

tl;dr - non-trivial, still in progress
